build-scripts/multidirectory.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+import os
+import shutil
+import subprocess
+import sys
+
+def remove_branches(remote):
+    branches = subprocess.check_output([
+        'git', 'branch', '--list', '--remotes', '--no-merged', 'HEAD',
+        remote + '/*'])
+    for branch in [x[len(remote) + 1:] for x in branches.split()]:
+        if not os.path.exists(os.path.join('jobs', branch, 'Jenkinsfile')):
+            subprocess.check_call([
+                'git', 'push', '--quiet', '--delete', remote, branch])
+
+def create_or_update_branches(remote):
+    if not os.path.exists('jobs/'):
+        return
+    msg = 'Auto-generated job branch from {{}} from {}'.format(
+        subprocess.check_output(['git', 'log', '-n', '1', '--pretty=%h']))
+    jobs = [d for (d, _, files) in os.walk('jobs/') if 'Jenkinsfile' in files]
+    for job in jobs:
```

Comment:
We should try to keep the level of abstraction high on this method so someone can take a cursory look at the script and understand what's going on:

```py
for branch in job_branches():
    if branch_exists_locally(branch):
        delete_local_branch(branch)

    create_local_branch(branch)
    push_branch(branch)
````